guides/auth-torii-with-github.md

```diff
@@ -0,0 +1,581 @@
+
+var ENV = {
+  ...
+
+  torii: {
+    sessionServiceName: 'session',
+    providers: {
+      'github-oauth2': {
+        apiKey: 'YOUR_API_KEY',
+        redirectUri: 'http://localhost:4200',
+        scope: 'repo user'
+      }
+    }
+  }
+};
+```
+
+`YOUR_API_KEY` is the client ID supplied when you registered your GitHub application.
+
+However, this API key is probably specific to your development environment, as is the value given for `redirectUri`
+above. This is fine while your app is in development, but as your application approaches production, you'll need to
+break some of it out.
+
+Additionally, it can be considered bad practice to distribute all of your API keys in your source code. Let's add a
+little complexity and address this concern by using `ember-cli-dotenv` to put our key in an external file that will be
+injected at build time. Install the package
+
+```js
+ember install ember-cli-dotenv
+```
+
+and create a `.env` file in the root of your project like
+
+```
+GITHUB_DEV_CLIENT_ID=<YOUR_API_KEY>
+```
+
+replacing `<YOUR_API_KEY>` with the client ID from your application registration.
+
+Break the configuration into the stage-specific parts of your `environment.js` like so.
+
+```js
+  var ENV = {
+    ...
+
+    torii: {
+      sessionServiceName: 'session',
+      providers: {
+        'github-oauth2': {
+          scope: 'repo user'
+        }
+      }
+    }
+  };
+
+  if(environment === 'development') {
+    ENV.torii.providers['github-oauth2'].apiKey = process.env.GITHUB_DEV_CLIENT_ID;
+    ENV.torii.providers['github-oauth2'].redirectUri = 'http://localhost:4200';
+  }
+```
+
+The values in the `development` block are the ones specific to that stage. You may choose to reuse those for the `test`
+stage, but you will need different ones for `production` as at least the API key will change and you will almost
+definitely host the application under a domain name. We will add to the `development` block later.
+
+### The Basic App
+
+Let's give ourselves an app to secure. We'll keep it simple. We'll have a login route for authentication and use the
+application index route to display the authenticated user's data, an operation that gives a 401 if you aren't
+authenticated.
+
+```
+ember g route application
+ember g controller application
+ember g route index
+ember g route login
+```
+
+Also, now is a good time to remove `ember-welcome-page` from your `package.json`.
+
+Let's put some content in our templates. First, our `application.hbs` will give a friendly greeting and some useful
+information so we know things are working as we proceed.
+
+```handlebars
+<!-- app/templates/application.hbs -->
+
+<h1>Welcome to the Demo App!</h1>
+
+API Key: {{config.apiKey}} <br />
+
+{{#if session.isAuthenticated}}
+    Authenticated
+{{else}}
+    Unauthenticated
+{{/if}}
+
+<hr />
+
+{{outlet}}
+```
+
+The `session` service is automatically injected, so there's no code for that. We'll add the `config` entry for now to
```

Comment:
I would change it now already so that people get used to explicitly inject the service instead of relying on the auto-injection. The advantage of explicit injection (`session: Ember.inject.service`) is simply that it's easier to follow and less magic.